### PR TITLE
Migrate generateNewXML* to use StringValue

### PR DESCRIPTION
**Pull request by riyafa**
## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/AbstractArrayValue.java [hunk 1]
```diff
@@ -392,5 +392,10 @@ public Object next() {
         public boolean hasNext() {
             return cursor < length;
         }
+
+        @Override
+        public StringValue bStringValue() {
```
Comment (vinok88) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/AbstractArrayValue.java]: Use BString

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/BmpStringValue.java [hunk 2]
```diff
@@ -59,6 +59,11 @@ public String stringValue() {
         return value;
     }
 
+    @Override
+    public StringValue bStringValue() {
+        return null;
```
Comment (vinok88) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/BmpStringValue.java]: 'null' return??

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/DecimalValue.java [hunk 3]
```diff
@@ -230,6 +230,11 @@ public String stringValue() {
         return value.toString();
     }
 
+    @Override
+    public StringValue bStringValue() {
```
Comment (vinok88) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/DecimalValue.java]: Use BString

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ErrorValue.java [hunk 4]
```diff
@@ -54,51 +54,52 @@
 
     private static final long serialVersionUID = 1L;
     private final BType type;
-    private final String reason;
+    private final StringValue reason;
```
Comment (vinok88) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ErrorValue.java]: Use BString

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/BmpStringValue.java (hunk 2 above)
Comment (riyafa) [reply bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/BmpStringValue.java]: Created an issue to track this: https://github.com/ballerina-platform/ballerina-lang/issues/20886 Need to migrate all implementations of BValue. Currently it returns null.

File: bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/StreamValue.java [hunk 5]
```diff
@@ -98,6 +98,11 @@ public String stringValue() {
         return "stream <" + getType().toString() + ">";
     }
 
+    @Override
+    public StringValue bStringValue() {
```
Comment (vinok88) [inline bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/StreamValue.java]: Use BString. Check other places as well.

**Pull request merged**